Detectors/CTP/workflowLumi/src/RawToLumiConverterSpec.cxx

```diff
@@ -0,0 +1,154 @@
+    if (linkCRU == o2::ctp::GBTLinkIDIntRec) {
+      payloadCTP = o2::ctp::NIntRecPayload;
+    } else {
+      return;
+    }
+    LOG(debug) << "RDH FEEid: " << feeID << " CTP CRU link:" << linkCRU << " Orbit:" << triggerOrbit;
+    // TF in 128 bits words
+    gsl::span<const uint8_t> payload(it.data(), it.size());
+    gbtword80_t gbtWord = 0;
+    int wordCount = 0;
+    std::vector<gbtword80_t> diglets;
+    if (orbit0 != triggerOrbit) {
+      // create lumi per HB
+      lumiPoint lp;
+      lp.counts = counts;
+      lp.ir.orbit = triggerOrbit;
+      mOutputLumiPoints.push_back(lp);
+      //
+      remnant = 0;
+      size_gbt = 0;
+      orbit0 = triggerOrbit;
+      counts = 0;
+    }
+    for (auto payloadWord : payload) {
+      // LOG(info) << wordCount << " payload:" <<  int(payloadWord);
+      if (wordCount == 15) {
+        wordCount = 0;
+      } else if (wordCount > 9) {
+        wordCount++;
+      } else if (wordCount == 9) {
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+        diglets.clear();
+        // LOG(info) << " gbtword:" << gbtWord;
+        o2::ctp::reco_workflow::RawToDigitConverterSpec::makeGBTWordInverse(diglets, gbtWord, remnant, size_gbt, payloadCTP);
+        // count tvx
+        for (auto diglet : diglets) {
+          // LOG(info) << " diglet:" << diglet;
+          gbtword80_t pld = (diglet & mTVXMask);
+          if (pld.count() != 0) {
+            counts++;
+          }
+        }
+        gbtWord = 0;
+      } else {
+        // std::cout << "wordCount:" << wordCount << std::endl;
+        for (int i = 0; i < 8; i++) {
+          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+        }
+        wordCount++;
+      }
+    }
+  }
+  LOG(info) << "[CTPRawToLumiConverter - run] Writing " << mOutputLumiPoints.size() << " lumiPoints ...";
+  ctx.outputs().snapshot(o2::framework::Output{"CTP", "LUMI", 0, o2::framework::Lifetime::Timeframe}, mOutputLumiPoints);
+}
+o2::framework::DataProcessorSpec o2::ctp::lumi_workflow::getRawToLumiConverterSpec(bool askDISTSTF)
+{
+  std::vector<o2::framework::InputSpec> inputs;
+  inputs.emplace_back("TF", o2::framework::ConcreteDataTypeMatcher{"CTP", "RAWDATA"}, o2::framework::Lifetime::Optional);
+  if (askDISTSTF) {
+    inputs.emplace_back("stdDist", "FLP", "DISTSUBTIMEFRAME", 0, o2::framework::Lifetime::Timeframe);
+  }
+
+  std::vector<o2::framework::OutputSpec> outputs;
+  outputs.emplace_back("CTP", "LUMI", 0, o2::framework::Lifetime::Timeframe);
+
+  return o2::framework::DataProcessorSpec{
+    "CTP-RawStreamDecoder",
+    inputs,
+    outputs,
+    o2::framework::AlgorithmSpec{o2::framework::adaptFromTask<o2::ctp::lumi_workflow::RawToLumiConverterSpec>()},
+    o2::framework::Options{{"result-file", o2::framework::VariantType::String, "/tmp/lumiCTPDecodeResults", {"Base name of the decoding results files."}}}};
```

Comment:
I don't see where this used. In any case, you should not write files on the FLP.

Reply:
LIne 153 removed.